### PR TITLE
website: support hidden pages in nav-data

### PR DESCRIPTION
**Pull request by zchsh**
👀 [Preview](https://vagrant-git-fork-zchsh-zssupport-hidden-pages-hashicorp.vercel.app)
🎟️ [Asana task](https://app.asana.com/0/1100423001970639/1200197752405255/f)

This PR is a follow-up to #12322, and implements a more permanent and clean solution for "hidden" pages, via updates to our `docs-page` component.

---

Items in `nav-data.json` now support an optional `"hidden": true` property, which will hide the item (and any descendants) from the docs sidebar.

Alongside this change, all `.mdx` files in each `content` folder must be included in the corresponding `nav-data` file. The changes to `docs-page` included an automated check which will throw an error during the website build process if there are any unlinked content files.

### CODE REVIEW

File: .github/workflows/check-unlinked-content.js [hunk 1]
```diff
@@ -1,141 +0,0 @@
-var fs = require("fs");
```
Comment (zchsh) [inline .github/workflows/check-unlinked-content.js:1]: This check is no longer needed, it has been implemented in our `docs-page` component as part of [react-components#220](https://github.com/hashicorp/react-components/pull/220).

File: website/components/_temp-enable-hidden-pages/index.js [hunk 2]
```diff
@@ -1,87 +0,0 @@
-// Imports below are used in server-side only
```
Comment (zchsh) [inline website/components/_temp-enable-hidden-pages/index.js:1]: This was a temporary hot fix for hidden pages, has been replaced by changes in [react-components#220](https://github.com/hashicorp/react-components/pull/220).

File: website/data/docs-nav-data-hidden.json [hunk 3]
```diff
@@ -1,32 +0,0 @@
-[
```
Comment (zchsh) [inline website/data/docs-nav-data-hidden.json:1]: This `*-nav-data-hidden.json` file was part of the temporary solution, so removing them. These nav-data items have been added to the proper `nav-data` file, with the `"hidden"` prop in place to reflect current state.

File: website/package.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
     "@hashicorp/nextjs-scripts": "18.1.0",
     "@hashicorp/react-button": "^5.0.1",
     "@hashicorp/react-code-block": "^4.0.1",
-    "@hashicorp/react-docs-page": "^13.1.4",
+    "@hashicorp/react-docs-page": "13.1.7-alpha.9",
```
Comment (zchsh) [inline website/package.json:11]: Will bump this to a stable release once these PRs are finalized and https://github.com/hashicorp/react-components/pull/220 is merged.

**Review by jescalan — DISMISSED** [review]

File: website/package.json (hunk 4 above)
Comment (zchsh) [reply website/package.json]: ✅ Bumped to stable

**Review by jescalan — APPROVED** [review]

**Review by BRKalow — APPROVED** [review]

**Pull request merged**